File: insertion_sort/Python/insertion_sort.py
def insertion_sort(arr):
    """
    Fuction to do insertion sort.

    :param arr: A list of element to sort.
    """
    for slot in range(1, len(arr)):
        value = arr[slot]
        test_slot = slot - 1
        while test_slot > -1 and arr[test_slot] > value:
            arr[test_slot + 1] = arr[test_slot]
            test_slot = test_slot - 1
        arr[test_slot + 1] = value

    return arr


def main():
    arr = [6, 5, 4, 3, 2, 1]
    print('Sorted element using Insertion Sort: {}'.format(
              ' '.join(map(str, insertion_sort(arr)))))


if __name__ == '__main__':
    main()

File: insertion_sort/Java/InsertionSort.java
class InsertionSort {
    /**
     * Insertion sort proceduce. Sorts the array in increasing
     * order in place. Time - O(n2), space O(1)
     * @param arr the array to be sorted.
     */
    static int[] insertionSort(int[] arr) {
        int i = 1;
        while (i < arr.length) {
            int key = arr[i];
            int j = i - 1;
            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j -= 1;
            }
            arr[j + 1] = key;
            i += 1;
        }
        return arr;
    }

    public static void main(String[] args) {
        int[] arr = {
            5,
            1,
            7,
            11,
            4,
            1
        };
        int[] sorted_arr = insertionSort(arr);
        for (int each_num: sorted_arr) {
            System.out.print(each_num + " ");
        }
    }
}
